### ISSUE TITLE
Not displaying all transactions after migrating to 4.8.0.2

### ISSUE THREAD

**Issue by frequentsailor** [firefly-iii/firefly-iii#2472]
I am running Firefly III version 4.8.0.2

**Description**
I tried migrating from my current installation (Version 4.7.17) on my NAS (created by using composer) to a new docker installation (using docker-compose) on my Raspberry Pi. I am running a MySQL Database in both cases. I dumped my MySQL Database from the old installation in an .sql file with phpMyAdmin ( MariaDB 10.3.11) and imported it on the new database using Adminer (MySQL 5.5.60-0+deb7u1).
Firefly starts up no problem and is running a few commands. After that I am taken to the homepage and everything looks good. Except I can not see all the transactions I entered on the old system. If I check a specific account I only get transactions listed up to a certain point. Strange thing is the changes in account balance in the graph also represent the missing transactions not shown in the list.

<img width="598" alt="Balance Graph" src="https://user-images.githubusercontent.com/54443289/63596679-0c284500-c5bc-11e9-8b55-88ebe4774724.png">

<img width="992" alt="Transactions" src="https://user-images.githubusercontent.com/54443289/63596685-121e2600-c5bc-11e9-8186-0b61f38669e4.png">

It would be great if you could help me figuring this out. Thanks!

**Extra info**

Debug information generated at 2019-08-23 15:26:09 Europe/Amsterdam for Firefly III version **4.8.0.2**.

| Variable | Content |
| --- | --- |
| FF version | 4.8.0.2 |
| FF API version | 0.10.0 |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | , daily |
| PHP version | 7.2.21 |
| Display errors | Off |
| Session start | 2019-08-01 00:00:00 |
| Session end | 2019-09-30 00:00:00 |
| Session first | 2019-03-01 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | apache2handler |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| Is Docker (.env) | false |
| bunq uses sandbox | false |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.1.2 Safari/605.1.15 |
| Loaded extensions | Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, bcmath, gd, intl, ldap, memcached, pdo_mysql, pdo_pgsql, sodium, Zend OPcache |

**Comment by JC5** [comment]
OK, so after a specific date the transactions are missing?

1. Re-import the backed-up database.
2. Remove all files from /storage/logs
3. un the following commands.

```
php artisan migrate --seed
php artisan firefly-iii:upgrade-database --force
```

1. Post the output from the command line
2. Post what you find in the logs. (errors etc)

**Comment by frequentsailor** [comment]
Thank you for the quick reply. Some steps took me a while to figure out. Here is the output of the commands:

`php artisan migrate --seed`

> pi@raspberrypi:~ $ docker-compose exec firefly_iii_app php artisan migrate --seed
Nothing to migrate.
Seeding: AccountTypeSeeder
Seeding: TransactionCurrencySeeder
Seeding: TransactionTypeSeeder
Seeding: PermissionSeeder
Seeding: LinkTypeSeeder
Seeding: ConfigSeeder
Database seeding completed successfully.
>
`php artisan firefly-iii:upgrade-database --force`

>pi@raspberrypi:~ $ docker-compose exec firefly_iii_app php artisan firefly-iii:upgrade-database --force
Now executing firefly-iii:transaction-identifiers
All split journal transaction identifiers are correct.
Verified and fixed transaction identifiers in 0.11 seconds.
Now executing firefly-iii:migrate-to-groups
Forcing the migration.
Found no split transaction journals. Nothing to do.
Migrate split journals to groups in 0.08 seconds.
Going to convert 123 transaction journals. Please hold..
Migrate all journals to groups in 4.34 seconds.
Migrated 123 transaction journal(s).
Now executing firefly-iii:account-currencies
All accounts are OK.
Verified and fixed account currencies in 0.32 seconds.
Now executing firefly-iii:transfer-currencies
All transfers have correct currency information.
Verified and fixed currency information for transfers in 4.23 seconds.
Now executing firefly-iii:other-currencies
Verified 805 transaction(s) and journal(s).
Verified and fixed transaction currencies in 7.09 seconds.
Now executing firefly-iii:migrate-notes
No notes to migrate.
Migrated notes in 0.02 seconds.
Now executing firefly-iii:migrate-attachments
All attachments are OK.
Migrated attachment notes in 0.01 seconds.
Now executing firefly-iii:bills-to-rules
All bills are OK.
Verified and fixed bills in 0.04 seconds.
Now executing firefly-iii:bl-currency
All budget limits are correct.
Verified budget limits in 0.03 seconds.
Now executing firefly-iii:cc-liabilities
No incorrectly stored credit card liabilities.
Now executing firefly-iii:back-to-journals
Forcing the command.
Check 52 transaction journal(s) for budget info.
Check 123 transaction journal(s) for category info.
Updated category and budget info for all transaction journals in 4.66 seconds.
Now executing firefly-iii:rename-account-meta
Renamed 2 account meta entries (entry).
Fixed account meta data in 0.07 seconds.
Now executing firefly-iii:fix-piggies
     The "--force" option does not exist.  
                                        
After running this everything seems to be normal. I can see all transactions now. This is quite surprising to me as this is basically what I did before :-)

Thank you very much for your help! I appreciate it very much.

**Comment by JC5** [comment]
Glad to hear it works :+1:
